[#42]refactor: Update firmwareSimulator and proto files

Mythois opened the pull request:
update the firmwareSimulator to handle habits of type timer. Also, update proto, desc and pb2.py files to match new protocol buffer message format

## Related issues
Issue #

## Summary

<!-- Provide a concise summary "Why are the changes needed"?  -->

## Changes Made

<!-- Describe the specific changes that have been made in this pull 
request. Provide details on the approach taken to address the problem 
and any notable implementation details. -->

## Checklist

- [x] I have added comments to code where needed
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have tested that there are no new bugs
- [ ] I have added tests that prove my fix is effective or that my feature works (Optional before testing is discovered)

EskildVeren commented on firmwareSimulator/habitTracker.py:
```diff
@@ -127,37 +127,74 @@ def on_lifecycle_connection_failure(self, lifecycle_connection_failure: mqtt5.Li
+                    case _:
+                        raise Exception("The provided format must either be JSON or proto_buff")
+            
+            case "timer":
```
rename "timer" to time

Mythois replied:
Should be fixed now. 
But best to have a new look at my code XD

EskildVeren commented on firmwareSimulator/habitTracker.py:
```diff
@@ -127,37 +127,74 @@ def on_lifecycle_connection_failure(self, lifecycle_connection_failure: mqtt5.Li
+                message = {
+                    "device_timestamp": int(round(datetime.timestamp(datetime.now()))),
+                    "habit_id": habit_id,
+                    "start_timestamp": int(round(datetime.timestamp(datetime.now()))),
```
Make changes so start isnt after stop

Mythois replied:
Should be fixed now. 
But best to have a new look at my code XD

Mythois commented:
I have rewritten the publish_message method to make it cleaner. While doing so I have incorporated your feedback. 
Minor changes have been done to other parts of the code in order for the rewrite of the publish_message method to work as intended.

I have also checked that the refactored method still works with AWS MQTT as before.
